check_patroni/types.py

```diff
@@ -76,6 +76,26 @@ def rest_api(self, service: str) -> Any:
                 return None
         raise nagiosplugin.CheckError("Connection failed for all provided endpoints")
 
+    def is_streaming_state_available(self) -> bool:
```

Comment:
The method could be cached, using `functools.cache()` or `lru_cache()`  depending on target Python version. This way, the caller would not need to keep the `has_streaming_state` variable for this.